### PR TITLE
added CANFD support

### PR DESCRIPTION
**Pull request by superuserx**
**Checklist:**

-   [ ] If you are new to Scapy: I have checked [CONTRIBUTING.md](https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md) (esp. section submitting-pull-requests)
-   [ ] I squashed commits belonging together
-   [ ] I added unit tests or explained why they are not relevant
-   [ ] I executed the regression tests (using `cd test && ./run_tests` or `tox`)
-   [ ] If the PR is still not finished, please create a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

fixes #xxx

### CODE REVIEW

**Comment by superuserx** [comment]
@polybassa huhu :)

**Comment by polybassa** [comment]
What's up? Please fix your unit test in order to go forward. 
-- 
Viele Grüße

Nils Weiß

On September 26, 2022 9:45:14 AM GMT+02:00, superuserx ***@***.***> wrote:
***@***.*** huhu :)

File: scapy/layers/can.py [hunk 1]
```diff
@@ -559,7 +577,11 @@ def read_packet(self, size=CAN_MTU):
         data = data.replace(b' ', b'')
         data = data.strip()
 
-        pkt = CAN(identifier=int(idn, 16), data=binascii.unhexlify(data))
```
Comment (polybassa) [inline scapy/layers/can.py:562]: You need to unhexlify the data

Comment (polybassa) [reply scapy/layers/can.py:562]: Please use ```scapy.compat.hex_bytes```

File: scapy/layers/can.py [hunk 2]
```diff
@@ -559,7 +577,12 @@ def read_packet(self, size=CAN_MTU):
...
+        pkt = CAN(identifier=int(idn, 16), data=hex_bytes(data))
```
Comment (polybassa) [inline scapy/layers/can.py:580]: one line later, you are overwriting pkt.

Comment (polybassa) [reply scapy/layers/can.py:580]: fix it in 582 and 584

**Comment by superuserx** [comment]
https://github.com/secdev/scapy/blob/3d5a61f541e082f28e51a8e25f78cfb2917d0829/scapy/layers/can.py#L158

Read PCAP file test fails because this line overwrites the registered DLT value of CAN type.
But there is no linktype value defined for CANFD by tcpdump.org (yet?).

**Comment by polybassa** [comment]
Closed, because of incompleteness

**Pull request closed without merging**